### DISCUSSION TITLE
Has anyone used devdojo/Voyager as a project for Laravel development

### DISCUSSION THREAD

**Discussion by liemarr** [thedevdojo/voyager#5877] [Q&A] [answered]
Opening the developed program on the mobile end takes several tens of seconds to load (in the case of poor network), and it also takes 4-5 seconds when the network is good. I found that tcg \ voyager \ publishable \ assets \ js \ app. js compressed 120000 lines of code before compression, but after compression, it still has 2.54MB, which is too heavy. I couldn't find the official documentation on how to use this in the production environment (in the case of mobile network), and there are not many JS interactive functions in the backend management. Is there a way to disassemble it, such as recording the jq itself or specifying the page to load.

I am a novice, that's why I have this question. I saw someone asking, and the official explanation is that：
It's only used for Voyager, not for your app, so it's REALLY not that bad. Having said that, yeah, we know it's a bit bloated. It's currently holding all of the JS dependencies in one file (jquery and more), which is why it appears so large.

There isn't too much we could do without breaking out the dependencies into their own file (then we'll have something like a 1MB file, and a 0.6MB file, which doesn't save you all that much). It'll take a complete top-down review and refactor to change that, which isn't something we're prepared to do right this moment, but we certainly appreciate your concern.

**Comment by liemarr** [comment, marked as answer]
I have already learned how to repackage myself, but I still find it a bit troublesome. A good solution is to divide it into 2-3 JS files, such as commonly used ones, those for editing interfaces, and those needed for advanced backend editing. This is the best solution. Now, I have rewritten the view file myself and only introduced the required JS files. Sorry for taking up discussion space. But this may be a rookie's process

**Reply by webtechlabs** [reply]
I totally agree about your separation of JS loading for what you are actively working on.

Have you considered doing a PR with your changes?  Better packaging and efficiency of a smaller must loading JS bundle would help everyone.  Thanks in advance.
